addons/account/models/account_report.py

```diff
@@ -280,6 +280,17 @@ def _validate_formula(self):
                     expression.report_line_id.display_name,
                 ))
 
+    @api.constrains('sequence')
+    def _validate_account_report_line(self):
+        # by default the sequence is 0 for the account report lines.
+        sequence_with_zero = self.report_id.line_ids.filtered(lambda l: l.sequence == 0)
+        if len(sequence_with_zero) == 1:
+            parent_set = set()
+            for line in self.report_id.line_ids.sorted(lambda l: l.sequence):
+                if line.parent_id and line.parent_id not in parent_set:
+                    raise ValidationError(_("Line %s must be after its parent %s.", line.name, line.parent_id.name))
```

Comment:
Put '"" around the line names. Also, the error message should be clearer

something like this ? =>  Line "%s" define line "%s" as its parent, but appears before it in the report. The parent must always come first.